code/obj/submachine/seed.dm

```diff
@@ -826,18 +831,34 @@
 
 		..()
 
+	proc/remove_distant_beaker()
+		// borgs and people with item arms don't insert the beaker into the machine itself
+		// but whenever something would happen to the dispenser and the beaker is far it should disappear
+		if(src.inserted && BOUNDS_DIST(src.inserted, src) > 0)
+			if (src.inserted == src.extract_to) src.extract_to = null
+			src.inserted = null
+			src.UpdateIcon()
+
 	proc/tryInsert(var/obj/item/W, var/mob/user)
-		if (isrobot(user))
-			boutput(user, "This machine does not accept containers from robots!")
+		remove_distant_beaker()
+
+		if(BOUNDS_DIST(user, src) > 0) // prevent message from appearing in case a borg inserts from afar
 			return
+
 		if(src.inserted)
 			boutput(user, "<span class='alert'>A container is already loaded into the machine.</span>")
 			return
 		src.inserted =  W
-		user.drop_item()
-		W.set_loc(src)
-		if(!src.extract_to) src.extract_to = W
-		boutput(user, "<span class='notice'>You add [W] to the machine!</span>")
+
+		if(!W.cant_drop)
+			user.drop_item()
+			if(!W.qdeled)
+				W.set_loc(src)
+		if(W.qdeled)
+			W = null
```

Comment:
not really sure what this is meant to do?

Reply:
Used the same lines of code Chemistry-Dispenser uses for checking if the item is not in the deletion queue.
https://github.com/goonstation/goonstation/blob/2fa63d2883a495084c87b3b6b62ab56689c8c3e9/code/modules/chemistry/Chemistry-Dispenser.dm#L132-L138

Reply:
might be some cursed reference stuff then, I'll assume it exists for a reason